src/ui/Dockerfile

```diff
@@ -1,22 +1,64 @@
-# Node Alpine Docker Image
-FROM node:18.9-alpine3.16
+###################
+# 1. BUILD
+###################
 
-# Variables
+# NPM 8.19.1 Alpine linux image
+FROM node:18.15-alpine3.16 AS build
+
+# Build variables
 ARG GITHUB_SHA
+ARG PORT
 ENV GITHUB_SHA=$GITHUB_SHA
+ENV PORT=$PORT
 
-# Install packages
-RUN apk add bash openrc openssh curl
-RUN rm -rf /var/cache/apk/*
-
-# Node setup
 WORKDIR /app
 
-COPY package.json .
-RUN npm i --omit=dev --legacy-peer-deps
+# Copy UI
+COPY --chown=node:node . .
+RUN npm ci --legacy-peer-deps --ignore-scripts
 RUN npm cache clean --force
 
-COPY . .
+# Build with all dependencies
+RUN npm run build
+
+# Lean NPM - Only install `dependencies`
+# `devDependencies` will still be resolved inside `package-lock.json`,
+# however they will not be installed inside `node_modules` directory.
+RUN npm ci --legacy-peer-deps --omit=dev --ignore-scripts
+
+###################
+# 2. PRODUCTION
+###################
+
+# NPM 9.5.1 Alpine linux image
+FROM node:19.8-alpine3.16 AS production
+
+WORKDIR /app
+
+# Copy from `build` to `prod`
+COPY --chown=node:node --from=build /app/package.json .
+COPY --chown=node:node --from=build /app/package-lock.json .
+COPY --chown=node:node --from=build /app/tsconfig.json .
+COPY --chown=node:node --from=build /app/node_modules ./node_modules
+COPY --chown=node:node --from=build /app/public ./public
+COPY --chown=node:node --from=build /app/scripts ./scripts
+COPY --chown=node:node --from=build /app/styles ./styles
+COPY --chown=node:node --from=build /app/server ./server
+COPY --chown=node:node --from=build /app/templates ./templates
+COPY --chown=node:node --from=build /app/types ./types
+
+# Non-root user
+USER node
+
+# Container healthcheck
```

Comment:
same as above, we shouldn't need this as Azure allows us to do it from settings and for local development we have it in a setting for docker-compose.yml file